Marksman/Refs.fs

```diff
@@ -316,17 +322,27 @@ module Dest =
                 | Dest.LinkDef _ -> [ srcDoc ]
                 | Dest.Heading _
                 | Dest.Doc _ -> Folder.docs folder |> List.ofSeq
+                | Dest.Tag _ -> Folder.docs folder |> List.ofSeq
 
             let referencingEls =
                 seq {
                     for targetDoc in targetDocs do
-                        let targetDocRefs = resolveLinks folder targetDoc
-
-                        let backRefs =
-                            findReferencingElements targetDocRefs declToFind
-                            |> Seq.map (fun (el, dest) -> targetDoc, el, dest)
-
-                        yield! backRefs
+                        match declToFind with
+                        | Dest.Tag(_, tag) ->
+                            let targets =
+                                Doc.index targetDoc
+                                |> Index.filterTagsByName tag.text
+                                |> Seq.map (fun t -> Dest.Tag(targetDoc, t))
+                                |> Seq.toArray
+                            targetDoc, T(tag), targets
```

Comment:
Almost there. This triple is a 'target doc' + 'element in the target doc' + array of dests.
However, here you're returning 'target doc' + T(tag) which is the original element in the source doc, which is incorrect.

You probably need something like this:
```fsharp
let targets =
    Doc.index targetDoc
    |> Index.filterTagsByName tag.data.name.text
    |> Seq.filter (fun t -> T t <> el)
    |> Seq.map (fun t -> targetDoc, T t, [| Dest.Tag(targetDoc, t) |])

yield! targets
```

Reply:
Now that you've explained it to me it makes total sense :see_no_evil: